### PR TITLE
Fix empty value on optional credentialsJSON for Spanner

### LINKED ISSUE #1941: Allow credential JSON for Spanner options
### Problem Statement

I am using SpiceDB library mode with Spanner for testing.
And I want to pass `CredentialsJSON` for Spanner.

### Solution Brainstorm

Add `credentialsJSON` for `spannerOptions`

I will send a PR.

### PR DESCRIPTION
**Pull request by lexcao**
Closes #1941
This is the follow-up PR for #1942, which fix the optional value issue.

### CODE REVIEW

**Review by vroldanbet — COMMENTED** [review]
The change makes sense to me, but concerned it has taken 3 PRs to get this right. Can we add a test?

File: internal/datastore/spanner/spanner.go [hunk 1]
```diff
@@ -141,14 +141,21 @@ func NewSpannerDatastore(ctx context.Context, database string, opts ...Option) (
...
+
 	client, err := spanner.NewClientWithConfig(context.Background(), database,
 		spanner.ClientConfig{SessionPoolConfig: cfg},
-		option.WithCredentialsFile(config.credentialsFilePath),
-		option.WithCredentialsJSON(config.credentialsJSON),
-		option.WithGRPCConnectionPool(max(config.readMaxOpen, config.writeMaxOpen)),
-		option.WithGRPCDialOption(
-			grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
-		),
+		append(optionalOpts,
```
Comment (josephschorr) [inline internal/datastore/spanner/spanner.go:152]: append this above the NewClient block

Comment (lexcao) [reply internal/datastore/spanner/spanner.go]: Done, thanks.

**Comment by lexcao** [comment]
Hi @vroldanbet 
I just thought it is options forwarding thing but I am not familiar with the code base at the beginning.
Is there any test case I can add?

**Comment by vroldanbet** [comment]
@lexcao I've made client initialization simpler by making the option handle empty credentials JSON, which Spanner client does not handle well. I also added a test. Please have a look and make sure this works for you.

**Comment by lexcao** [comment]
Hi @vroldanbet 
Thanks for your suggestions, but it would not resolve the issue.
The key point is `option.WithCredentialsJSON(config.credentialsJSON)` from Spanner, 
which should skip adding that option if the credentialsJSON is nil.

The internal operation in the `WithCredentialsJSON` method is making nil to `empty slice of bytes`, 
which would cause initialization error for Spanner with `empty slice`

https://github.com/googleapis/google-api-go-client/blob/e732ee3983555c5b0ca708033d9b8ec675f99019/option/option.go#L61-L70

**Comment by vroldanbet** [comment]
@lexcao, you are right. I'd suggest removing my commit, and if anything adding a new commit only with the test. I'd also suggest opening a PR upstream spanner SDK to fix this annoying behaviour

**Comment by lexcao** [comment]
Thanks, let me try to raise a PR for Spanner.

**Comment by lexcao** [comment]
Hi @vroldanbet 
I just raised a PR in the upstream. 
https://github.com/googleapis/google-api-go-client/pull/2643
Waiting for review.

For now, I think we should add the fix for this, since it could break the use of `CredentialsFile` for Spanner.
I have updated my code, you can review now.

I will make it more ergonomic after the upstream accepts my PR then.

**Review by vroldanbet — APPROVED** [review]

**Pull request merged**